### PR TITLE
GDScript: Include lambda dependencies

### PR DESCRIPTION
**Pull request by rune-scape**
theres no issue tracker, as the bug this fixes is very difficult to stumble on

- includes dependencies of lambdas to script dependencies
- also adds static initializer to dependencies

### CODE REVIEW

**Comment by rune-scape** [comment]
hmmm, the failed check seems like an unrelated crash...

File: modules/gdscript/gdscript.cpp [hunk 1]
```diff
@@ -1246,40 +1246,43 @@ GDScript *GDScript::_get_gdscript_from_variant(const Variant &p_variant) {
 	return Object::cast_to<GDScript>(obj);
 }
 
+void GDScript::_get_function_dependencies(GDScriptFunction *p_func, RBSet<GDScript *> &p_dependencies, const GDScript *p_except) {
```
Comment (adamscott) [inline modules/gdscript/gdscript.cpp:1249]: Is `_get_function_dependencies` the right name? It actually inserts dependencies, isn't it?
(we know that you actually call the existing `scr->_get_dependencies()`, which inserts dependencies, hence the name)

Comment (adamscott) [reply modules/gdscript/gdscript.cpp:1249]: @rune-scape That's funny, I'm the one that named `_get_dependencies()`.

**Review by adamscott — CHANGES_REQUESTED** [review]
The PR is excellent, nice catch @rune-scape. If you could rename the functions to what they do, as commented, it would be awesome.

File: modules/gdscript/gdscript.cpp (hunk 1 above)
Comment (vonagam) [reply modules/gdscript/gdscript.cpp:1249]: Personally, I would expect `fetch` to be a returning (non-void) function too. For me `_collect_dependencies` or `_fill_dependencies` reflect better what happens.

**Review by adamscott — APPROVED** [review]

**Pull request merged**